asf_core_data/getters/epc/data_batches.py

```diff
@@ -83,47 +104,111 @@ def get_all_batch_names(data_path="S3", rel_path=base_config.RAW_DATA_PATH):
 
     else:
 
-        data_path = get_batch_path(data_path / rel_path.parent, data_path=data_path)
-        batches = [p.name for p in data_path.glob("*/") if not p.name.startswith(".")]
+        data_path = get_batch_path(
+            data_path / rel_path, data_path=data_path, check_folder="input"
+        )
+        batches = [
+            p.name
+            for p in data_path.glob("*/")
+            if not p.name.startswith(".")
+            and not p.name.endswith(".zip")
+            and not p.name.startswith("__")
+        ]
 
     return batches
 
 
-def get_most_recent_batch(data_path=None, rel_path=base_config.RAW_DATA_PATH):
+def get_latest_hist_installers():
+    """Get the filename for the latest batch of historical MCS installer data.
+
+    Returns:
+        str: Latest filename for historical data.
+    """
+
+    bucket = "asf-core-data"
+    path = "outputs/MCS/installers"
+
+    batches = [key for key in data_getters.get_s3_dir_files(bucket, path)]
+
+    if not batches:
+        raise IOError("No suitable baches found.")
+
+    # Return highest value since files are marked with date stamps in format yyyymmdd
+    return max(batches)
+
+
+def get_latest_mcs_epc_joined_batch(version="most_relevant"):
+    """Get the filename for the latest batch of MCS-EPC joined data for given version.
+
+    Args:
+        version (str, optional): Which merging type is used for EPC. Defaults to most_relevant.
+            All options: most_relevant, newest, full.
+
+    Returns:
+        str: Latest EPC/MCS batch filename for given version.
+    """
+
+    bucket = "asf-core-data"
+    path = "outputs/MCS/"
+
+    batches = [
+        key for key in data_getters.get_s3_dir_files(bucket, path) if "old" not in key
+    ]
+    batches = [batch for batch in batches if batch[:-11].endswith(version)]
+
+    if not batches:
```

Comment:
It's actually quite common and considered best practice to check for non-empty lists with: `if list`. It will return False if the list has 0 elements and True if it has at least one element.

Reply:
interesting! Didn't know it was even considered best practice as I've always used `len()` :)